### PR TITLE
Logging util

### PR DESCRIPTION
**Pull request by cadmiumcat**
- Adds and implements logging utility
- Removes previous logging
- Removes broken link from responsive.css 
- Adds missing fonts
- [ ] There are more broken links on responsive.css - this needs fixing

### CODE REVIEW

File: blaise_survey_manager/sample_intake/functions.py [hunk 1]
```diff
@@ -182,38 +183,34 @@ def process_sample(request):
...
         populate_tables(sample_header_dict, sample_records)
     except ServerParkNotDefined:
-        logger.error('method: functions.process_sample, Server Park Not Defined')
         db.session.rollback()
         set_sample_header_status(sample_header, 'Server park not defined')
         return False
-    except Exception:
-        logger.error('method: functions.process_sample, Error uploading sample')
+    except Exception as ex:
+        log.info("Error uploading sample " + str(ex))
```
Comment (martyncolmer) [inline blaise_survey_manager/sample_intake/functions.py]: should be log.error

File: blaise_survey_manager/sample_intake/functions.py [hunk 2]
```diff
@@ -182,38 +183,34 @@ def process_sample(request):
...
     if not request.is_json:
         return False
 
     try:
         payload = json.loads(request.json)
-    except Exception:
-        logger.error('method: functions.process_feed_forward, Error loading json file')
-        print("Error loading json.")
+    except Exception as ex:
+        log.debug("Error loading JSON " + str(ex))
```
Comment (martyncolmer) [inline blaise_survey_manager/sample_intake/functions.py]: should be log.error

File: blaise_survey_manager/sample_intake/functions.py [hunk 3]
```diff
@@ -234,12 +231,12 @@ def process_feed_forward(request):
 
         return True
     except Exception:
-        logger.error('method: functions.process_feed_forward, Error loading proccessing feed forward')
+        log.info("Could not complete process feed forward data")
```
Comment (martyncolmer) [inline blaise_survey_manager/sample_intake/functions.py]: should be log.error

File: blaise_survey_manager/sample_intake/views.py [hunk 4]
```diff
@@ -65,18 +59,16 @@ def fragment_sample_files_table():
 @login_required
 @restrict_access(feature='sample_upload')
 def delete_sample_file():
-    logger.info('method: views.delete_sample_file')
     sample_header_id = request.args.get('sample_header_id', None, type=str)
     delete_sample(sample_header_id)
+    log.debug("Sample deleted - sample_header_id = " + str(sample_header_id))
```
Comment (martyncolmer) [inline blaise_survey_manager/sample_intake/views.py]: I'd promote this to log.info

File: blaise_survey_manager/sample_intake/views.py [hunk 5]
```diff
@@ -154,8 +138,9 @@ def manual_feed_forward_file_upload():
             payload['feed_forward_data'] = [row for row in reader]
 
             json_payload = json.dumps(payload)
-        except Exception:
-            logger.error('method: views.manual_feed_forward_file_upload, Error in CSV file')
+        except Exception as ex:
+            log.debug(ex)
```
Comment (martyncolmer) [inline blaise_survey_manager/sample_intake/views.py]: should be log.error

File: blaise_survey_manager/app/views.py [hunk 6]
```diff
@@ -10,7 +11,12 @@
 @main.route('/')
 @cache.cached(timeout=1000)
 def home():
-    return render_template('index.html')
+    log.info("Requesting Blaise Survey Manager home page")
+    try:
+        return render_template('index.html')
+    except Exception as e:
+        log.error(str(e))
```
Comment (salmaniqbal) [inline blaise_survey_manager/app/views.py]: Do what you did elsewhere
`log.error("Msg:" + e)`

File: blaise_survey_manager/case_creation/functions.py [hunk 7]
```diff
@@ -113,17 +116,20 @@ def deploy_cases(instrument_id):
                     payload[key] = value
             rabbit_payload['payload'] = payload
             if not send_to_rabbit_queue(case.ServerPark.name + '.case_creation', json.dumps(rabbit_payload), channel):
+                log.debug('Error deploying case: message queue error')
```
Comment (salmaniqbal) [inline blaise_survey_manager/case_creation/functions.py]: Can you see if there is some `Case` identifier info that you can log e.g. Case.name, Case.Id is probably better or some other malarkey

File: blaise_survey_manager/case_creation/functions.py [hunk 8]
```diff
@@ -149,7 +157,8 @@ def get_case_summary_by_instrument(instrument_id, page):
                                Case.allocation_status
                                ) \
         .filter(Case.instrument_id == instrument_id) \
-        .order_by(Case.id)\
+        .order_by(Case.id) \
         .paginate(page, 10, False)
+    log.debug('Providing summary by instrument')
```
Comment (salmaniqbal) [inline blaise_survey_manager/case_creation/functions.py]: Bang in the instrument id too please :pray:

File: blaise_survey_manager/sample_intake/functions.py [hunk 9]
```diff
@@ -3,42 +3,45 @@
...
 
 def sample_header_exists(survey_id, field_period_id, phase, status):
     # check if sample uploaded already
-    logger.info('functions.sample_header_exists')
     sample_header = SampleHeader.query.filter_by(survey_id=survey_id,
                                                  field_period_id=field_period_id,
                                                  phase=phase,
                                                  status=status).first()
     if sample_header:
+        log.info("Sample file already uploaded for"
```
Comment (salmaniqbal) [inline blaise_survey_manager/sample_intake/functions.py]: I like this kind of logging, good work @cadmiumcat :cat2:

File: blaise_survey_manager/sample_intake/functions.py [hunk 10]
```diff
@@ -54,20 +57,22 @@ def check_sample_header(survey_id, field_period_id, phase, status):
 
 
 def check_payload_schema(sample_header_dict):
-    logger.info('method:functions.check_payload_schema, check payload schema for survey_id, field_period_id, phase & payload')
     if 'survey_id' not in sample_header_dict:
+        log.warning("Missing survey_id in sample header")
```
Comment (salmaniqbal) [inline blaise_survey_manager/sample_intake/functions.py]: Please add the ids in the logs

File: blaise_survey_manager/sample_intake/functions.py [hunk 11]
```diff
@@ -54,20 +57,22 @@ def check_sample_header(survey_id, field_period_id, phase, status):
...
         return False
     if 'field_period_id' not in sample_header_dict:
+        log.warning("Missing field_period_id in sample header")
         return False
```
Comment (salmaniqbal) [inline blaise_survey_manager/sample_intake/functions.py]: Please add the ids in the logs (and for phase & payload too)

File: blaise_survey_manager/sample_intake/functions.py [hunk 12]
```diff
@@ -91,7 +96,7 @@ def populate_tables(sample_header, sample_records):
         case.issue_number = '1'
         case.household = '1'
         if case.mode not in server_parks_dict:
-            logger.error('method: functions.populate_tables, Server Park Not Defined')
+            log.warning("Server Park not defined")
```
Comment (salmaniqbal) [inline blaise_survey_manager/sample_intake/functions.py]: can this say something like 
`log.warning("Server Park:" + serverpark "not defined for case mode: " + case.mode)`

something like that, if you  get my drift :racing_car:

File: blaise_survey_manager/sample_intake/views.py [hunk 13]
```diff
@@ -35,7 +30,7 @@ def intake_json():
 @login_required
 @restrict_access(feature='sample_upload')
 def intake_feed_forward_json():
-    logger.info('method: views.intake_feed_forward_json, request: {}'.format(request))
+    # logger.info('method: views.intake_feed_forward_json, request: {}'.format(request))
```
Comment (salmaniqbal) [inline blaise_survey_manager/sample_intake/views.py]: You can delete this line, no need to comment it :scissors:

File: blaise_survey_manager/sample_intake/views.py [hunk 14]
```diff
@@ -88,35 +80,27 @@ def manual_sample_file_upload():
     phase_choices.extend(db.session.query(Phase.id, Phase.description).all())
     form.phase.choices = phase_choices
     if form.validate_on_submit():
-        # logger.info('method: views.manual_sample_file_upload, validonsubmit, request: {}'.format(request))
+        log.info("Starting manual sample upload")
```
Comment (salmaniqbal) [inline blaise_survey_manager/sample_intake/views.py]: Can you see if there is any identifiable information that might be useful to log. If not then its cool

File: blaise_survey_manager/survey_config/views.py [hunk 15]
```diff
@@ -89,6 +91,7 @@ def delete_survey():
     survey = Survey.query.filter_by(id=survey_id).one()
     db.session.delete(survey)
     db.session.commit()
+    log.info("Survey " + survey.survey_name + " deleted")
```
Comment (salmaniqbal) [inline blaise_survey_manager/survey_config/views.py]: I usually put colons before any variable that I am going to be printing

`log.info("Survey: " + survey.survey_name + " deleted")`

Makes it easier to spot in reams and reams of logs :newspaper_roll:

File: blaise_survey_manager/survey_config/views.py [hunk 16]
```diff
@@ -141,6 +145,7 @@ def delete_field_period():
     field_period = FieldPeriod.query.filter_by(id=field_period_id).one()
     db.session.delete(field_period)
     db.session.commit()
+    log.info("Field period " + field_period.stage + " deleted")
```
Comment (salmaniqbal) [inline blaise_survey_manager/survey_config/views.py]: :newspaper_roll:

File: blaise_survey_manager/survey_config/views.py [hunk 17]
```diff
@@ -244,6 +251,7 @@ def add_instrument():
         form.populate_obj(instrument)
         db.session.add(instrument)
         db.session.commit()
+        log.info("Instrument " + instrument.name + " added")
```
Comment (salmaniqbal) [inline blaise_survey_manager/survey_config/views.py]: :newspaper_roll:

File: blaise_survey_manager/survey_config/views.py [hunk 18]
```diff
@@ -280,5 +288,6 @@ def delete_instrument():
     instrument = Instrument.query.filter_by(id=instrument_id).one()
     db.session.delete(instrument)
     db.session.commit()
+    log.info("Instrument " + instrument.name + " deleted")
```
Comment (salmaniqbal) [inline blaise_survey_manager/survey_config/views.py]: :newspaper_roll:

File: blaise_survey_manager/users/forms.py [hunk 19]
```diff
@@ -41,11 +44,13 @@ def validate(self):
 
         # if our validators do not pass
         if not check_validate:
+            log.debug('Invalid user: ' + json.dumps(self.errors))
```
Comment (salmaniqbal) [inline blaise_survey_manager/users/forms.py]: Good logging :cake:

**Pull request merged**